packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -114,6 +114,12 @@ class ColdRunner extends ResidentRunner {
 
     appStartedCompleter?.complete();
 
+    if (debuggingOptions.vmserviceOutFile != null) {
+      final File vmserviceOutFile = fs.file(debuggingOptions.vmserviceOutFile);
+      vmserviceOutFile.createSync(recursive: true);
+      vmserviceOutFile.writeAsStringSync(flutterDevices.first.vmServices.first.wsAddress.toString());
```

Comment:
Is picking the first vmService arbitrary (not sure why a device would have more than one)?

Reply:
We don't support any debugging functionality for multiple devices, so by convention we always return the first vmservice address